lib/active_record_doctor/schema_cache.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+module ActiveRecordDoctor
+  class SchemaCache # :nodoc:
```

Comment:
I suggest using a different name here. The reasoning is that the callers of that class shouldn't care whether it's caching or not. Its primary role is schema reflection/introspection with caching as a side benefit.

My suggestion would be to:

- Rename this to `CachingSchemaInspector` (to make the class name indicate what this class is about).
- Rename all occurrences of `schema_cache` to `schema_inspector` (the callers don't need to know about caching).